src/TestUtils.js

```diff
@@ -26,3 +26,19 @@ export function destroyAllDataPermanently(fast) {
     })
   );
 }
+
+export function resolvingPromise() {
```

Comment:
Interesting, do you know why moving it from helper to src fixed this?

Reply:
It doesn't I put it in the wrong place in my last PR. Forgot a TestUtils file already exists.

The fix is a combination of using a global spy for better control, removing unnecessary configuration and improving the flaky test which I documented.

Reply:
Nice